packages/data/src/onboarding/actions.js

```diff
@@ -347,3 +369,21 @@ export function* hideTaskList( id ) {
 export function* optimisticallyCompleteTask( id ) {
 	yield optimisticallyCompleteTaskRequest( id );
 }
+
+export function* actionTask( id ) {
+	yield actionTaskRequest( id );
+
+	try {
+		const task = yield apiFetch( {
+			path: `${ WC_ADMIN_NAMESPACE }/onboarding/tasks/${ id }/action`,
+			method: 'POST',
+		} );
+
+		yield actionTaskSuccess(
+			possiblyPruneTaskData( task, [ 'isDismissed', 'isSnoozed' ] )
```

Comment:
Ooh one comment I forgot, we should only list `isActioned` in this array, you can remove the `isDismissed` and `isSnoozed`.